code/src/java/pcgen/gui2/converter/panel/WriteDirectoryPanel.java

```diff
@@ -191,42 +190,35 @@ private void showWarning()
 
 	private List<Campaign> getExistingPccs()
 	{
-		List<File> existingFiles = new ArrayList<>();
-		findPCCFiles(path, existingFiles);
+		List<Path> existingFiles = findPCCFiles(path);
 
 		List<Campaign> matchingCampaigns = new ArrayList<>();
 
 		for (Campaign camp : campaignList)
 		{
 			File campFile = new File(camp.getSourceURI());
-			for (File file : existingFiles)
+			if (existingFiles.stream()
+			                 .anyMatch(file -> file.getFileName().toString().equals(campFile.getName())))
 			{
-				if (file.getName().equals(campFile.getName()))
-				{
-					matchingCampaigns.add(camp);
-					break;
-				}
+				matchingCampaigns.add(camp);
 			}
 		}
 
 		return matchingCampaigns;
 	}
 
-	private void findPCCFiles(File aDirectory, List<File> existingFiles)
+	private static List<Path> findPCCFiles(File aDirectory)
 	{
-		if (aDirectory.isDirectory())
+		try
 		{
-			for (File file : aDirectory.listFiles(pccFileFilter))
-			{
-				if (file.isDirectory())
-				{
-					findPCCFiles(file, existingFiles);
-					continue;
-				}
-				existingFiles.add(file);
-			}
+			return Files.walk(aDirectory.toPath())
+			            .filter(file -> file.getFileName().toString().endsWith("pcc"))
```

Comment:
One of the earlier file -> file.blah items also did .toLowerCase.... do we intend to be case sensitive here?

Also, it would be interesting to see if some of these filters could be shared at some point (shouldn't hold up this PR)

Reply:
yeah, long term we should pull these out into a PCGenFileFilters holder class or something similar.